tools/get_coverage_ros2_pkg.py

```diff
@@ -0,0 +1,121 @@
+#!/usr/bin/env python3
+
+import argparse
+import os
+import requests
+import subprocess
+import sys
+import tempfile
+
+parser = argparse.ArgumentParser()
+parser.add_argument("jenkins_coverage_build", help="URL of a ci.ro2s.org build using coverage (i.e https://ci.ros2.org/job/ci_linux_coverage/182)")
+parser.add_argument("ros_package", help="ROS package name to get the coverage rate from (i.e: rcutils)")
+args = parser.parse_args()
+
+input_url = args.jenkins_coverage_build
+input_pkg = args.ros_package
+
+
+# Compare if a list is inside other list in order
+# From here: https://stackoverflow.com/a/20789669
+def is_slice_in_list(s, l):
+    len_s = len(s)
+    return any(s == l[i:len_s + i] for i in range(len(l) - len_s + 1))
+
+
+def create_colcon_workspace():
+    ros2_ws_path = os.path.join(tempfile.gettempdir(), 'ros2_coverage_tool')
```

Comment:
The temporary directory should be removed when the script doesn't need it anymore. I would suggest using `tempfile.TemporaryDirectory` with a context manager.

Reply:
d501288